### ISSUE TITLE
[bug] Propagation of cpp_info.name in cmake_find_package_multi

### ISSUE THREAD

**Issue by Morwenn** [conan-io/conan#5969]
The `cmake_find_package_multi` correctly generates *XXXConfig.cmake* files using the dependency's `cpp_info.name` for the `XXX` part, but doesn't always correctly use `cpp_info.name` inside the generated file.

### Environment Details
  * Operating System+version: Ubuntu 18.04
  * Conan version: 1.19.2
  * Python version: 3.6.8

### Steps to reproduce

I have got an ITK recipe that requires both `zlib` and `hdf5` and uses `cmake_find_package_multi`, the `hdf5` recipe also requires `zlib`. Everything worked smoothly up to yesterday when the `cpp_info.name` of `zlib` was changed to `ZLIB`: now `cmake_find_package_multi` correctly generates *HDF5Config.cmake* and *ZLIBConfig.cmake* as expected, but the *HDF5Config.cmake* file contains the following lines:

```cmake
include(CMakeFindDependencyMacro)
if(${CMAKE_VERSION} VERSION_LESS "3.9.0")
  find_package(zlib REQUIRED NO_MODULE)
else()
  find_dependency(zlib REQUIRED NO_MODULE)
endif()
get_target_property(tmp zlib::zlib INTERFACE_LINK_LIBRARIES)
```

When the `find_dependency` above is called, it searches a file called *zlibConfig.cmake* instead of the generated *ZLIBConfig.cmake*. I can't tell for sure, but I believe that `cmake_find_package_multi` doesn't correctly propagate the `cpp_info.name` in to the `find_dependency` and subsequent target names when it should.

### Logs (Executed commands with output) (Include/Attach if Applicable)

The relevant logs are the following ones:

```
CMake Error at /usr/share/cmake-3.10/Modules/CMakeFindDependencyMacro.cmake:48 (find_package):
  Could not find a package configuration file provided by "zlib" with any of
  the following names:

    zlibConfig.cmake
    zlib-config.cmake

  Add the installation prefix of "zlib" to CMAKE_PREFIX_PATH or set
  "zlib_DIR" to a directory containing one of the above files.  If "zlib"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  HDF5Config.cmake:43 (find_dependency)
  source_subfolder/Modules/ThirdParty/HDF5/itk-module-init.cmake:5 (find_package)
  source_subfolder/CMake/ITKModuleEnablement.cmake:318 (include)
  source_subfolder/CMakeLists.txt:433 (include)


-- Configuring incomplete, errors occurred!
```

**Comment by danimtb** [comment]
Sure, this is a clear bug. Thanks a lot for the detailed report

**Comment by danimtb** [comment]
Fixed by #5973
